File: cifrado/escritorio/codigo/Kotlin/src/alfabeto/Alfabeto.kt
package alfabeto

class Alfabeto(){
    val alfabeto = mutableListOf<Char>()

    init {
        llenar(32,127)
        llenar(161,172)
        llenar(173,256)
    }

    fun llenar(inicio:Int,final:Int){
        for (indice in inicio..final){
            alfabeto.add(indice.toChar())
        }
    }

    fun obtenerCaracter(posicion:Int): Char {
        if (posicion >= alfabeto.size)
            return alfabeto[(posicion-alfabeto.size)-1]
        else if (posicion < 0)
            return alfabeto[(alfabeto.size+posicion)+1]
        else
            return alfabeto[posicion]
    }

    fun obtenerPosicion(caracter:Char): Int{
        if (caracter in alfabeto)
            return alfabeto.indexOf(caracter)
        else
            return -1
    }

    fun obtenerTamnio(): Int {
        return alfabeto.size
    }
}

File: cifrado/escritorio/codigo/Ruby/alfabeto/Alfabeto.rb
class Alfabeto
    $alfabeto = []

    def initialize()
        (0..254).each do |x|
            if( x.chr(Encoding::UTF_8).gsub(/[^[:print:]]/i, '') != "" )
              $alfabeto.push(x.chr(Encoding::UTF_8) ) 
            end
        end
    end

    def obtenerCaracter(posicion)
        if posicion > $alfabeto.length
            return $alfabeto[(posicion-$alfabeto.length)-1]
        elsif posicion < 0
            return $alfabeto[(posicion+$alfabeto.length)+1]
        else
            return $alfabeto[posicion]
        end
    end 

    def obtenerPosicion(caracter)
        if $alfabeto.include?(caracter)
            return $alfabeto.index(caracter)
        else
            return -1
        end
    end

    def obtenerTamanio()
        return $alfabeto.length
    end
    
end